tests/framework/components/test_component.py

```diff
@@ -0,0 +1,252 @@
+from typing import List
+
+import pandas as pd
+
+from vivarium import Component, InteractiveContext
+from vivarium.framework.event import Event
+from vivarium.framework.population import SimulantData
+
+
+class ColumnCreator(Component):
+    def __repr__(self):
+        return "ColumnCreator()"
+
+    @property
+    def name(self) -> str:
+        return "column_creator"
+
+    @property
+    def columns_created(self) -> List[str]:
+        return ["test_column_1", "test_column_2", "test_column_3"]
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        initialization_data = pd.DataFrame(
+            {column: 9 for column in self.columns_created}, index=pop_data.index
+        )
+        self.population_view.update(initialization_data)
+
+
+class ColumnRequirer(Component):
+    def __repr__(self):
+        return "ColumnRequirer()"
+
+    @property
+    def name(self) -> str:
+        return "column_requirer"
+
+    @property
+    def columns_required(self) -> List[str]:
+        return ["test_column_1", "test_column_2"]
+
+
+class ColumnCreatorAndRequirer(Component):
+    def __repr__(self):
+        return "ColumnCreatorAndRequirer()"
+
+    @property
+    def name(self) -> str:
+        return "column_creator_and_requirer"
+
+    @property
+    def columns_required(self) -> List[str]:
+        return ["test_column_2"]
+
+    @property
+    def columns_created(self) -> List[str]:
+        return ["test_column_4"]
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        initialization_data = pd.DataFrame({"test_column_4": 8}, index=pop_data.index)
+        self.population_view.update(initialization_data)
+
+
+class AllColumnsRequirer(Component):
+    def __repr__(self):
+        return "AllColumnsRequirer()"
+
+    @property
+    def name(self) -> str:
+        return "all_columns_requirer"
+
+    @property
+    def columns_required(self) -> List[str]:
+        return []
```

Comment:
test_column_1-4 because of empty list correct?

Reply:
I'm not sure I understand your question, but the empty list here means that the component requires all columns.

Reply:
Yes was verifying that.